bpf/lib/nat.h

```diff
@@ -723,8 +729,15 @@ snat_v4_needs_masquerade(struct __ctx_buff *ctx __maybe_unused,
 		 * rp_filter=1.
 		 */
 
-		if (remote_ep->flag_skip_tunnel)
+		if (remote_ep->flag_skip_tunnel) {
+			#ifdef ENABLE_REMOTE_NODE_SNAT
+			{
+				target->addr = IPV4_MASQUERADE;
+				return NAT_NEEDED;
+			}
+			#endif
 			return NAT_PUNT_TO_STACK;
+		}
```

Comment:
How about adding the `ENABLE_REMOTE_NODE_SNAT` block just once, at the top of the whole section?

Are you planning to add the same handling in the IPv6 path?

Reply:
I did not want to make change for the Tunnel mode case that is already in place. For IPv6 We probably need to introduce a new config filed since at least for Azure we are not NAtting traffic to IPv6 in dualstack mode.

Reply:
I see. I think you'll want to describe in a lot more detail what this new feature flag is doing exactly, and in which scenarios it is useful / ineffective. `Documentation/network/concepts/masquerading.rst` is probably a good resting place.

Reply:
Will do

Reply:
> For IPv6 We probably need to introduce a new config filed since at least for Azure we are not NAtting traffic to IPv6 in dualstack mode.

I don't understand how that matters to the new flag. We already have `enableIPv6Masquerade`, so if you don't want to NAT traffic to IPv6, then that can be disabled. It doesn't need to be tied in any way to how remote nodes are treated, does it?

Reply:
@julianwiedmann 
Per @pchaigno comment, I actually made use of the same flag to SNAT traffic to IPv6 as well since we already have V6Masqurade in configmap. I also took the condition out of Tunnel mode. Since if this flag is set it wil be an explicit request for masqurading and will trump the tunneling mode condition.
I made a new commeit to address this

Reply:
> It doesn't need to be tied in any way to how remote nodes are treated, does it?

Agree. There shouldn't be any difference with respect to ipv4/ipv6. the new flag should be common for both. the intention of this flag is to keep behavior same as that of upstream ipmasq agent.

Reply:
The issue has been addressed.